### PR TITLE
Add unit tests for dropwizard.util.StringsTest

### PR DESCRIPTION
**Pull request by paulkennethkent**
###### Problem:
I've analysed your code base and noticed that `dropwizard.util.StringsTest` is not fully tested.

###### Solution:
I've written some tests for the functions in this class with the help of [Diffblue](https://www.diffblue.com/) [Cover](https://www.diffblue.com/products). Hopefully, these tests should help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other particular classes that you consider important.

### CODE REVIEW

File: dropwizard-util/src/test/java/io/dropwizard/util/StringsTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  @Test
+  public void emptyToNull() {
+    Assert.assertEquals("A",  Strings.emptyToNull("A"));
+    Assert.assertNull(Strings.emptyToNull(""));
+  }
+
+  @Test
+  public void isNullOrEmpty() {
+    Assert.assertEquals(false, Strings.isNullOrEmpty("A"));
```
Comment (freddytuxworth) [inline dropwizard-util/src/test/java/io/dropwizard/util/StringsTest.java]: use assertTrue + assertFalse

File: dropwizard-util/src/test/java/io/dropwizard/util/StringsTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import org.junit.rules.ExpectedException;
+
+public class StringsTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void emptyToNull() {
+    Assert.assertEquals("A",  Strings.emptyToNull("A"));
+    Assert.assertNull(Strings.emptyToNull(""));
```
Comment (freddytuxworth) [inline dropwizard-util/src/test/java/io/dropwizard/util/StringsTest.java]: Try with `null` input?

File: dropwizard-util/src/test/java/io/dropwizard/util/StringsTest.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  @Test
+  public void isNullOrEmpty() {
+    Assert.assertEquals(false, Strings.isNullOrEmpty("A"));
+    Assert.assertEquals(true, Strings.isNullOrEmpty(""));
+  }
+
+  @Test
+  public void nullToEmpty() {
+    Assert.assertEquals("", Strings.nullToEmpty(""));
+    Assert.assertEquals("", Strings.nullToEmpty(null));
```
Comment (freddytuxworth) [inline dropwizard-util/src/test/java/io/dropwizard/util/StringsTest.java]: Try with non-empty input?

**Review by JohnLBergqvist — APPROVED** [review]

**Review by freddytuxworth — APPROVED** [review]

File: dropwizard-util/src/test/java/io/dropwizard/util/StringsTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      Assert.assertEquals("A",  Strings.emptyToNull("A"));
+      Assert.assertNull(Strings.emptyToNull(""));
+      Assert.assertNull(Strings.emptyToNull(null));
+  }
+
+  @Test
+  public void isNullOrEmpty() {
+      Assert.assertFalse(Strings.isNullOrEmpty("A"));
+      Assert.assertTrue(Strings.isNullOrEmpty(""));
+  }
```
Comment (peterschrammel) [inline dropwizard-util/src/test/java/io/dropwizard/util/StringsTest.java:23]: Add `Assert.assertTrue(Strings.isNullOrEmpty(null));`

**Review by peterschrammel — APPROVED** [review]

**Review by JohnLBergqvist — APPROVED** [review]

**Review by freddytuxworth — APPROVED** [review]

**Pull request closed without merging**